duckduckgo/Android
# Move pixel API to its own gradle lib module

Task/Issue URL: https://app.asana.com/0/414730916066338/1199931182139284/f
Tech Design URL: 
CC: 

**Description**:
This PR moves the `statistics` package, that contains the ATB and Pixel APIs to its own gradle lib module. The PR:

- Moved statistics package to its own module
- Added dagger as implementation to use Inject annotations
- Added common gradle module and moved WorkerInjectorPlugin to it
- Moved AppUrl to common module and fix pixel service and entity
- Moved Device.kt to common module and fix VERSION_NAME access
- Moved exception package to common module
- Removed default Build.VERSION_NAME from UncaughtExceptionEntity
- Fixed the AppInstallationReferrerStateListener<>AtbInitializer dependency
- Fixed tests

I have left the corresponding test in the `app` gradle module for now to be sure the are executed in CI (I was not sured whether CI was prepared to execute tests in other gradle lib modules).
Once that's done, we can move the tests as well


**Steps to test this PR**:
1. install/upgrade from this branch
2. open the app test basic features work, bookmarks, settings, fire proofing sites, tabs, etc.
3. verify pixels are sent normally


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
